lib/status-bar/index.js

```diff
@@ -0,0 +1,83 @@
+/* @flow */
+
+import { CompositeDisposable } from 'sb-event-kit'
+import disposify from 'disposify'
+
+import Element from './element'
+import { $file } from '../helpers'
+import type { LinterMessage } from '../types'
+
+export default class StatusBar {
+  element: Element;
+  messages: Array<LinterMessage>;
+  subscriptions: CompositeDisposable;
+  statusBarRepresents: 'Entire Project' | 'Current File';
+  statusBarClickBehavior: 'Toggle Panel' | 'Jump to next error';
+
+  constructor() {
+    this.element = new Element()
+    this.messages = []
+    this.subscriptions = new CompositeDisposable()
+
+    this.subscriptions.add(this.element)
+    this.subscriptions.add(atom.config.observe('linter-ui-default.statusBarRepresents', statusBarRepresents => {
+      const shouldUpdate = typeof this.statusBarRepresents !== 'undefined'
```

Comment:
Rename this to `notInitial` or something similar to make it clearer that this is a check for the initial call to this callback vs. a setting change.